src/Microsoft.Health.Client/ManagedIdentityCredentialProvider.cs

```diff
@@ -14,27 +14,33 @@ namespace Microsoft.Health.Client
 {
     public class ManagedIdentityCredentialProvider : CredentialProvider
     {
-        private readonly ManagedIdentityCredentialConfiguration _managedIdentityCredentialConfiguration;
+        private readonly IOptionsMonitor<ManagedIdentityCredentialConfiguration> _managedIdentityCredentialConfigurationMonitor;
         private readonly IHttpClientFactory _httpClientFactory;
+        private readonly string _optionsName;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ManagedIdentityCredentialProvider"/> class.
         /// This class is used to obtain a token for the configured resource via Managed Identity.
         /// </summary>
-        /// <param name="managedIdentityCredentialConfiguration">The configuration of the token to obtain.</param>
+        /// <param name="managedIdentityCredentialConfigurationMonitorMonitor">The configuration of the token to obtain.</param>
         /// <param name="httpClientFactory">An optional <see cref="IHttpClientFactory"/> for use within the <see cref="AzureServiceTokenProvider"/>.</param>
-        public ManagedIdentityCredentialProvider(IOptions<ManagedIdentityCredentialConfiguration> managedIdentityCredentialConfiguration, IHttpClientFactory httpClientFactory = null)
+        /// <param name="optionsName">Optional name to use when retrieving options from the IOptionsMonitor</param>
+        public ManagedIdentityCredentialProvider(IOptionsMonitor<ManagedIdentityCredentialConfiguration> managedIdentityCredentialConfigurationMonitorMonitor, IHttpClientFactory httpClientFactory = null, string optionsName = null)
         {
-            EnsureArg.IsNotNull(managedIdentityCredentialConfiguration?.Value, nameof(managedIdentityCredentialConfiguration));
+            EnsureArg.IsNotNull(managedIdentityCredentialConfigurationMonitorMonitor, nameof(managedIdentityCredentialConfigurationMonitorMonitor));
 
-            _managedIdentityCredentialConfiguration = managedIdentityCredentialConfiguration.Value;
+            _managedIdentityCredentialConfigurationMonitor = managedIdentityCredentialConfigurationMonitorMonitor;
+            _optionsName = optionsName;
             _httpClientFactory = httpClientFactory;
         }
 
         protected override async Task<string> BearerTokenFunction(CancellationToken cancellationToken)
         {
+            ManagedIdentityCredentialConfiguration managedIdentityCredentialConfiguration = string.IsNullOrEmpty(_optionsName)
```

Comment:
If we normalize empty string to `null` in the ctor, we can simply pass `null` to `Get(string)` to get the `CurrentValue`, based on what I see in their [source code](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Microsoft.Extensions.Options/src/OptionsMonitor.cs#L70).